src/explore-education-statistics-frontend/next.config.js

```diff
@@ -3,19 +3,105 @@ const flowRight = require('lodash/fp/flowRight');
 const withTranspileModules = require('next-transpile-modules');
 const path = require('path');
 
+const cspConnectSrc = [
+  "'self'",
+  process.env.NEXT_PUBLIC_CONTENT_API_BASE_URL.replace('/api', ''),
+  process.env.NEXT_PUBLIC_DATA_API_BASE_URL.replace('/api', ''),
+  process.env.NEXT_PUBLIC_NOTIFICATION_API_BASE_URL.replace('/api', ''),
+  'https://www.google-analytics.com',
+  'https://dc.services.visualstudio.com/v2/track',
+];
+
+const cspScriptSrc = [
+  "'self'",
+  'https://www.google-analytics.com/',
+  "'unsafe-inline'",
+  "'unsafe-eval'",
+];
+
+const frameScriptSrc = [
+  "'self'",
+  'https://department-for-education.shinyapps.io/',
+  'https://dfe-analytical-services.github.io/',
+];
+
+const contentSecurityPolicy = `
+  default-src 'self';
+  script-src ${cspScriptSrc.join(' ')};
+  style-src 'self' 'unsafe-inline';
+  img-src * blob: data: https://www.google-analytics.com/;
+  media-src 'self' blob: data:;
+  font-src 'self';
+  connect-src ${
+    process.env.NODE_ENV !== 'production' ? '*' : cspConnectSrc.join(' ')
+  };
+  frame-src ${frameScriptSrc.join(' ')};
+  frame-ancestors 'self';
+  child-src 'self';
+`;
+
+const securityHeaders = [
```

Comment:
Where have we actually got these headers from? 

We seem to be missing a bunch of headers and defaults set by Helmet - see [their website](https://helmetjs.github.io/).

If we're replacing Helmet, I would expect us to at least have all the defaults set (plus our overrides).

Reply:
I think for now we'll keep the previous helmet CSP in server.js. I initially moved the CSP stuff into `next.config.js` when I assumed we might try to get rid of the custom server and just use the default server Next provides out of the box